### PR TITLE
exposing bam partition size to GATKSparkTool

### PR DESCRIPTION
**Pull request by lbergelson**
fix for #1064

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSource.java [hunk 1]
```diff
@@ -84,9 +96,20 @@ public ReadsSparkSource(JavaSparkContext ctx) {
...
      */
     public JavaRDD<GATKRead> getParallelReads(final String bam) {
+        return getParallelReads(bam, DEFAULT_SPLIT_SIZE);
+    }
+
+    /**
+     * Loads Reads using Hadoop-BAM. For local files, bam must have the fully-qualified path,
+     * i.e., file:///path/to/bam.bam. This excludes unmapped reads.
+     * @param bam file to load
+     * @param splitSize
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSource.java]: Document the `splitSize` parameter here

**Comment by droazen** [comment]
:+1: merge when ready

**Comment by laserson** [comment]
Is there some discussion about why this is necessary?  This doesn't seem like the right place to expose this to me.  Wouldn't this make more sense being set, for example, on the CLI with the `spark-submit` command?

**Comment by droazen** [comment]
See https://github.com/broadinstitute/gatk/issues/961 and comment in https://github.com/broadinstitute/gatk/issues/1070 -- an alpha goal is to make hellbender spark tools easier to run by providing a front-end to invoke `spark-submit` / `gcloud dataproc` on the user's behalf with sensible defaults for all important options. So we don't anticipate that running `spark-submit` manually will be the typical use case.

Having said that, if a user does run `spark-submit` manually and specifies `conf` options there, we should ensure that these override any hellbender defaults in `SparkCommandLineProgram`

**Pull request merged**